# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: js/src/jit-test/tests/gc/bug-1654186.js
# Commit: 1937b90d16a4
# Full Hash: 1937b90d16a4714abbd46473d5b9ff76c4435ed8
# Author: Jon Coppeard <jcoppeard@mozilla.com>
# Date: 2020-07-22 04:32:41
# Description:
#   Bug 1654186 - Call onOutOfMallocMemory() on OOM in GCRuntime::decommitFreeArenas so that we release protected free areans before trying to decommit them r=sfink
#   
#   Compacting temporarily mprotects released arenas in debug builds to catch problems. This interferes with Chunk::decommitFreeArenasWithoutUnlocking which checks for free arenas by looking at every arena in the chunk. This is different to Chunk::decommitOneFreeArena which uses the list of free arenas.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D84334
# ==============================================================================

diff -r 654c839a370a -r 1937b90d16a4 js/src/jit-test/tests/gc/bug-1654186.js
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/js/src/jit-test/tests/gc/bug-1654186.js	Tue Jul 21 15:27:41 2020 +0000
@@ -0,0 +1,8 @@
+// |jit-test| allow-oom; skip-if: !('oomAfterAllocations' in this)
+
+gczeal(14, 5);
+var g = newGlobal();
+g.eval("(" + function() {
+    oomAfterAllocations(100);
+} + ")()");
+f.x("");
